dataactvalidator/config/sqlrules/c9_award_financial.sql

```diff
@@ -1,14 +1,22 @@
-SELECT afa.row_number, afa.fain, afa.uri
+SELECT DISTINCT afa.row_number, afa.fain, afa.uri
 FROM award_financial_assistance AS afa
 WHERE afa.submission_id = {0}
-	AND CAST(afa.federal_action_obligation as numeric) > 0
-	AND afa.row_number NOT IN (
-		SELECT afa.row_number
-		FROM award_financial_assistance AS afa
-			JOIN award_financial AS af
-				ON (afa.fain IS NOT DISTINCT FROM af.fain
-				AND afa.uri IS NOT DISTINCT FROM af.uri
-				AND afa.submission_id = af.submission_id)
-		WHERE CAST(afa.federal_action_obligation as numeric) > 0
-			AND afa.submission_id = {0}
-	);
\ No newline at end of file
+    AND COALESCE(CAST(afa.federal_action_obligation as numeric),0) <> 0
+    AND COALESCE(CAST(afa.original_loan_subsidy_cost as numeric),0) <> 0
+    AND (
+        afa.row_number NOT IN (
+            SELECT afa.row_number
+            FROM award_financial_assistance AS afa
+                JOIN award_financial AS af
+                    ON (afa.submission_id = af.submission_id
+                        AND (afa.fain IS NOT DISTINCT FROM af.fain
+                            AND afa.uri IS NOT DISTINCT FROM af.uri)
+                    )
+            WHERE COALESCE(CAST(afa.federal_action_obligation as numeric),0) <> 0
+                AND COALESCE(CAST(afa.original_loan_subsidy_cost as numeric),0) <> 0
+                AND afa.submission_id = {0}
+        ) OR (
+            afa.fain IS NOT NULL
+            AND afa.uri IS NOT NULL
```

Comment:
Two questions on this.  First, should FAIN and URI both present get a warning for C8 and a warning for C9, or should these be consolidated in some way?  Second, should the C9 warning for FAIN and URI both present be skipped when the fields above are equal to zero?  From the text "Unique FAIN/URI from file D2 exists in file C, except D2 records where FederalActionObligation and OriginalLoanSubsidyCost = 0", I get the impression that exception only applies to the check between C and D2, not the FAIN and URI both provided warning.